db/fetcher/indexer_iterators.go

```diff
@@ -45,45 +45,52 @@ const (
 // For example, iteration over condition _eq and _gt will have completely different logic.
 type indexIterator interface {
 	Init(context.Context, datastore.DSReaderWriter) error
-	Next() (core.IndexDataStoreKey, bool, error)
+	Next() indexIterResult
 	Close() error
 }
 
+type indexIterResult struct {
+	key      core.IndexDataStoreKey
+	foundKey bool
+	value    []byte
+	err      error
+}
+
 type queryResultIterator struct {
 	resultIter query.Results
 }
 
-func (i queryResultIterator) Next() (core.IndexDataStoreKey, bool, error) {
+func (i *queryResultIterator) Next() indexIterResult {
```

Comment:
suggestion: I would leave the error as a seperate return value. This makes it more obvious to the caller that the error needs to be checked.